### DISCUSSION TITLE
Multitask GP - Inference on new data

### DISCUSSION THREAD

**Discussion by RossDF** [cornellius-gp/gpytorch#2101] [General]
Hi,

I have a few hundred time-series observations across 100 days from the same feature at different locations. Each of those observations differ slightly from each other by follow the same general trend.
From these I'm trying to build a model to forecast a second set of incomplete data (50 days worth) to estimate their change over time over the missing 50 days.

Is a MultitaskGPModel correct for this application? If yes, how could I actually input that incomplete data into a trained model and receive the forecast?
I did find a single similar question that someone has asked in the past (https://github.com/cornellius-gp/gpytorch/issues/985) however I can't seem to make that work.

**Comment by gpleiss** [comment]
I don't think I understand exactly what you're asking. Can you define what "same feature at different locations," "differ slightly from each other," and "incomplete data" mean?

I think a Multitask GP model is exactly what you're looking for, and the [Multitask GP tutorial](https://docs.gpytorch.ai/en/stable/examples/03_Multitask_Exact_GPs/Multitask_GP_Regression.html) should show you how to make predictions on previously unseen test inputs.

**Reply by RossDF** [reply]
Sorry, that could've been more concise.
Let's say I was monitoring water quality at 100 different locations along a waterway; I have 70 y * 100 x, and 30 y * 50 x.
I want to infer the missing 50 x on the second set, based on a model trained on the first set.

That tutorial shows how to test the model on a new x for the same y dataset, but not how to substitute a new dataset for y (and not retrain the model).

**Reply by gpleiss** [reply]
What do you mean by "trained"? Do you simply want to reuse the hyperparameters, or do you actually want to make predictions on the second set conditioned on the data in your first set?

If it's the latter, then what is the relationship between the `y` values in the first set and the second set? For example, if the 30y in the second set have no relation to the 70y in the first set, then conditioning on the first set shouldn't affect the predictions. Alternatively, if the 30y in the second set are are subset of the 70y in the first set, then you could just make predictions on all 70y from the first set and then just select the 30y that you actually care about.

**Reply by RossDF** [reply]
Sorry gpleiss,
I would be wanting to just reuse those hyperparameters from the first trained model. https://github.com/cornellius-gp/gpytorch/issues/985 What is described by grishabhg is pretty much my exact use case.

**Reply by gpleiss** [reply]
With your trained model you can then call

```python
model.set_train_data(second_x, second_y)
model.eval()
# ... make predictions
```

or alternatively

```python
# first_set_model (instance of MyModelClass) is trained on first_x, first_y
second_set_model = MyModelClass(second_x, second_y)
second_set_model.load_state_dict(first_set_model.state_dict())
second_set_model.eval()
# ... make predictions
```
